app/src/main/java/com/abisoft/todocompose/ui/viewModel/addTaskViewModel/AddTaskViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.abisoft.todocompose.ui.viewModel.addTaskViewModel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.abisoft.todocompose.TodoItemsRepository
+import com.abisoft.todocompose.model.ImportanceNetwork
+import com.abisoft.todocompose.model.TodoItemNetwork
+import com.abisoft.todocompose.model.TodoItemPost
+import com.example.myapplication.okhttp_client.TodoDto
+import kotlinx.coroutines.launch
+import java.util.UUID
+
+
+class AddTaskViewModel(private val repository: TodoItemsRepository) : ViewModel() {
+
+        private val _todoList = MutableLiveData<List<TodoItemNetwork>>()
+        val todoList: LiveData<List<TodoItemNetwork>> get() = _todoList
+
+    fun addNewTask(task: TodoDto) {
+        viewModelScope.launch {
+            try {
+                val currentRevision = repository.getCurrentRevision()
+
+                // TodoDto obyektini TodoItemNetwork ga o'zgartirish
+                val newTaskNetwork = TodoItemNetwork(
```

Comment:
You shouldn't work with network classes in ViewModel. ViewModel is a presentation layer and Network is a data layer. It should have low coupling